services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/team/TeamManagerImpl.java

```diff
@@ -730,6 +726,7 @@ public String getIconPreviewURL(UserInfo userInfo, String teamId) throws NotFoun
 	}
 
 	String getFileHandleId(String teamId) throws NotFoundException {
+		teamDAO.validateTeamExists(teamId);
```

Comment:
You are making an additional query that is not necessary, the get(teamId) already throws an exception.